### PR TITLE
perf(ui): replace clsx and tailwind-merge with cn

### PR DESCRIPTION
**Pull request by Orlandohub**
## Summary
- Swap the `cn()` helper from `clsx` + `tailwind-merge` to shadcn's [`cn`](https://github.com/shadcn-ui/cn) engine (`0.2.4`), a drop-in replacement with the same merge semantics.
- Announcement: https://x.com/shadcn/status/2095179046904607092
- Call sites are unchanged; they still import `cn` from `@/libs/utils/utils`.
- `clsx` remains a transitive dependency of `class-variance-authority` and `react-tweet` (react-tweet default-imports it, so it is not aliased to `cn`).

## Benchmark
Isolated-process microbench on this machine, using class strings from this app (Button, gutter, cards). Best of 5 runs, 200,000 iters per scenario, `clsx` + `tailwind-merge` vs `cn`.

| scenario | clsx + tailwind-merge | cn | faster |
| --- | ---: | ---: | ---: |
| typical component (stable literals) | 953.5 ns | 12.5 ns | 76.4× |
| same classes as last render (cache hit) | 923.6 ns | 17.1 ns | 53.9× |
| gutter override (breakpoint groups) | 96.3 ns | 4.3 ns | 22.2× |
| conditional + object (card/sidebar) | 290.6 ns | 59.8 ns | 4.9× |
| cold unique strings (SSR-style) | 839.4 ns | 30.3 ns | 27.7× |

The big win is repeated component calls with stable string identities, which is the shape almost every `cn()` call in this repo has.

## Test plan
- [ ] `npx vitest run --project unit src/libs/utils/utils.test.ts src/components/atoms/Button/Button.test.tsx`
- [ ] Spot-check landing and sign-in: buttons, header, and dialog class overrides still look correct
- [ ] Confirm conflicting utilities still resolve as before (`p-4` + `p-2` → `p-2`; gutter `px-4 lg:px-6 xl:px-0` + `px-0` only replaces the base group)

### CODE REVIEW

File: src/libs/utils/utils.ts [hunk 1]
```diff
@@ -14,9 +12,7 @@ import type {
   GetDisplayTagsOptions,
 } from './utils.types';
 
-export function cn(...inputs: ClassValue[]) {
-  return twMerge(clsx(inputs));
-}
+export { cn } from 'cn';
```
[the remark this answers is not in the document]
Comment (Orlandohub) [reply src/libs/utils/utils.ts:15]: Intentional. `cn()` already lived in `utils.ts`; this is the drop-in swap so ~184 call sites stay put. The no-barrel rule is about `index.ts` aggregators, not this existing helper.

**Review by secondl1ght — APPROVED** [review]
Approving. The swap is a faithful drop-in and the claims in the description hold up.

**Verified**

- **Output parity.** Extracted every literal class string from all 418 `cn()` and 17 `cva()` calls in `src` and compared `twMerge(clsx(...))` at the removed versions (`clsx@2.1.1`, `tailwind-merge@3.6.0`) against `cn@0.2.4` on per-call argument combinations, seeded random pairs/triples across the pool, and ~250 hand-picked edge cases (variants, `!` modifier, arbitrary values, container queries, custom theme colors, the gutter override from the test plan). 212,595 inputs, zero mismatches.
- **Cache semantics.** The argument-identity cache resolves objects/arrays to strings before matching, so mutating a reference between calls behaves the same as before. All caches are bounded, so no SSR memory-growth concern.
- **Provenance.** Published by shadcn from GitHub Actions with npm provenance attestations since 0.2.1. The `cn` name previously belonged to an unrelated 2013 package (0.1.x), so the exact `0.2.4` pin matters and is in place.
- **Compatibility.** Engines require Node 20+; the repo pins 24. No remaining `clsx` / `tailwind-merge` / `twMerge` / `ClassValue` references in `src`, and there was no `extendTailwindMerge` config to migrate. The vitest run from the test plan passes locally (270 tests); CI is green on the head commit.

**Non-blocking notes**

1. **Bundle size goes up slightly.** Minified is a wash, but the compiled tables gzip worse than tailwind-merge's config. `clsx` stays in the bundle via `class-variance-authority` either way, so the net client delta is roughly the difference below. Might be worth a line in the description so "perf" reads as runtime rather than payload.

   | bundle | min | min+gz |
   | --- | ---: | ---: |
   | clsx + tailwind-merge 3.6.0 | 27.4 KB | 8.5 KB |
   | cn 0.2.4 | 27.4 KB | 10.8 KB |

2. **Lockfile version drift.** The 1.5.0 → 1.8.0 bump in `package-lock.json` is just catching up to `package.json` on `dev`. Correct, only unrelated.
3. **Stale reference.** `.cursor/skills/code-review/SKILL.md` line 174 still describes `cn()` as clsx+tailwind-merge.
4. **Guard against reintroduction.** Since `clsx` remains hoisted, a direct `import { clsx } from 'clsx'` would silently work as a phantom dependency. Adding `clsx` and `tailwind-merge` to the existing `no-restricted-imports` patterns in `eslint.config.mjs` would close that.
5. **Test coverage.** The `cn` tests in `utils.test.ts` cover joining and a single `p-4`/`p-2` conflict but not the per-variant-group override that the `layoutClasses.ts` comment relies on. One assertion for `px-4 lg:px-6 xl:px-0` + `px-0` would pin it.
6. **Maturity.** The package is two days old at 0.x. The exact pin and the parity results make that acceptable; just expect some churn on future bumps.
7. **Greptile thread.** Agree with the resolution: the no-barrel rule targets `index.ts` aggregators, and keeping the `@/libs/utils/utils` import path for 184 files is the right call.

---
_Generated by [Claude Code](https://claude.ai/code)_
